stan/math/prim/fun/exp.hpp

```diff
@@ -62,6 +64,45 @@ inline auto exp(const Eigen::MatrixBase<Derived>& x) {
   return x.derived().array().exp().matrix().eval();
 }
 
+namespace internal {
+/**
+ * Return the natural (base e) complex exponentiation of the specified
+ * complex argument.
+ *
+ * @tparam V value type (must be Stan autodiff type)
+ * @param z complex number
+ * @return natural exponentiation of specified complex number
+ * @see documentation for `std::complex` for boundary condition and
+ * branch cut details
+ */
+template <typename V>
+inline std::complex<V> complex_exp(const std::complex<V>& z) {
+  if (is_inf(z.real()) && z.real() > 0) {
+    if (is_nan(z.imag()) || z.imag() == 0) {
+      // (+inf, nan), (+inf, 0)
+      return z;
+    } else if (is_inf(z.imag()) && z.imag() > 0) {
+      // (+inf, +inf)
+      return {z.real(), std::numeric_limits<double>::quiet_NaN()};
+    } else if (is_inf(z.imag()) && z.imag() < 0) {
+      // (+inf, -inf)
+      return {std::numeric_limits<double>::quiet_NaN(),
```

Comment:
Comment doesn't match implementation

Reply:
What doesn't match?  Line 88 is the (+inf, -inf) result, because the real part's tested to be +inf in the containing conditional.